iothub_client/src/iothub_client_dll.def

```diff
@@ -145,6 +145,12 @@ EXPORTS
     IoTHubMessage_SetInputName
     IoTHubMessage_SetMessageId
     IoTHubMessage_SetProperty
+	
+	AMQP_Protocol
+	AMQP_Protocol_over_WebSocketsTls
+	HTTP_Protocol
+	MQTT_Protocol
+	MQTT_WebSocket_Protocol
```

Comment:
These will have to be broken up in 3 different files so if someone runs cmake without MQTT or AMQP or HTTP it doesn't break.  So you'll need to create 3 iothub_transport_XXXX.def files and then include each in the cmake depending on the transports that are being used.